apps/web/src/features/spaces/components/SpaceAddressBook/DeleteContactDialog.tsx

```
@@ -16,15 +22,38 @@ type DeleteContactDialogProps = {
 }
 
 const DeleteContactDialog = ({ name, address, networks, onClose }: DeleteContactDialogProps) => {
+  const [error, setError] = useState<string>()
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const dispatch = useAppDispatch()
+  const spaceId = useCurrentSpaceId()
+  const [deleteEntry] = useAddressBooksDeleteByAddressV1Mutation()
+
   const handleConfirm = async () => {
+    setError(undefined)
+
     try {
-      // TODO: Implement the API call to remove the address book entry
-      //   await Promise.all(entryIds.map((id) => removeAddressBookEntry(id)))
-      console.log('remove address book entries', name, address, networks)
+      setIsSubmitting(true)
       trackEvent({ ...SPACE_EVENTS.REMOVE_ADDRESS_SUBMIT })
+      const response = await deleteEntry({ spaceId: Number(spaceId), address })
+
+      if (response.error) {
+        setError('Something went wrong deleting the contact. Please try again.')
+        return
```

I think we should set isSubmitting to false here. Alternatively, we could throw an error in this case to trigger the catch block, which would handle both setError and setIsSubmitting. WDYT?

---

I think because it is inside the finally block it will always be reached even if we return early here but I also see the duplication. Ideally we can show a more descriptive error if `result.error` and show a generic error if the whole request throws but currently this is missing on the CGW.

---

True, you're right! I missed that the finally block will be executed despite the early return.